Add header blocks to `setup.py` and `runtests.py`

lukpueh opened the pull request.

Commit:
Add header block to setup.py

Commit:
Add header block to runtests.py

SantiagoTorres commented on setup.py:
````diff
@@ -1,6 +1,33 @@
+
+  # Installation from GitHub
+    ```
+    pip install git+git://github.com/in-toto/in-toto@develop
````
interesting that `git+git` exists as a protocol. I think I'd rather opt for `git+https`.

lukpueh replied:
Will change

SantiagoTorres replied:
Just a note, I wasn't being pedantic, apparently you can do git+git :P

lukpueh replied:
I think https is a good choice. 
Btw. here's a good [overview of `git`'s protocols](https://git-scm.com/book/id/v2/Git-on-the-Server-The-Protocols) which are [all supported by `pip`](https://pip.pypa.io/en/stable/reference/pip_install/#git).

SantiagoTorres commented on setup.py:
```diff
@@ -1,6 +1,33 @@
+<Purpose>
+  setup.py script to install in-toto framework and in-toto scripts
+
+  # System Dependencies
```
Our *system* doesn't depend on git, or pip per se. I think this is from the readme, but we could probably call these "development requirements" rather than "system dependencies"? What do you think?

lukpueh replied:
Makes sense to not list those under "system dependencies". What about the oxymoronic "optional requirements"? Because at least pip is not really a "development requirement" either.

SantiagoTorres replied:
I think probably going with debian's `recommended tools` for pip/git, and the others as actual `requirements`. What do you think?

lukpueh replied:
Sounds good to me, I'd vote for adding `virtualenvs` to recommended tools as well

SantiagoTorres replied:
sounds good 👍

Commit:
Add "recommended tools" section to setup header

Following @SantiagoTorres' suggestion this commit adds a
"recommended tools" section to setup.py's header block to list
pip, git and virtualenv, which are not actually system
dependencies

Commit:
Change pip over git installation protocol to https

Following @SantiagoTorres' comment this commit changes the pip
over git installation instruction to use https instead of the git
protocol.

See https://git-scm.com/book/id/v2/Git-on-the-Server-The-Protocols
for pros and cons of git protocols and
https://pip.pypa.io/en/stable/reference/pip_install/#git for a list
of protocols supported by pip.

lukpueh commented:
Thanks for reviewing, @SantiagoTorres!
